src/sentry/sentry_apps/api/endpoints/sentry_app_requests_v2.py

```diff
@@ -0,0 +1,178 @@
+from dataclasses import dataclass
+from datetime import datetime, timezone
+
+from dateutil.parser import parse as parse_date
+from rest_framework import serializers, status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import control_silo_endpoint
+from sentry.api.paginator import GenericOffsetPaginator
+from sentry.api.serializers import serialize
+from sentry.models.organizationmapping import OrganizationMapping
+from sentry.sentry_apps.api.bases.sentryapps import SentryAppBaseEndpoint, SentryAppStatsPermission
+from sentry.sentry_apps.api.serializers.request_v2 import RequestSerializer
+from sentry.sentry_apps.models.sentry_app import SentryApp
+from sentry.sentry_apps.services.app_request import RpcSentryAppRequest, SentryAppRequestFilterArgs
+from sentry.sentry_apps.services.app_request.serial import serialize_rpc_sentry_app_request
+from sentry.sentry_apps.services.app_request.service import app_request_service
+from sentry.types.region import find_all_region_names
+from sentry.utils.sentry_apps import EXTENDED_VALID_EVENTS, SentryAppWebhookRequestsBuffer
+
+
+class IncomingRequestSerializer(serializers.Serializer):
+    date_format = "%Y-%m-%d %H:%M:%S"
+    eventType = serializers.ChoiceField(
+        choices=EXTENDED_VALID_EVENTS,
+        required=False,
+    )
+    errorsOnly = serializers.BooleanField(required=False)
+    organizationSlug = serializers.CharField(required=False)
+    start = serializers.DateTimeField(
+        format=date_format,
+        default=datetime.strptime("2000-01-01 00:00:00", date_format).replace(tzinfo=timezone.utc),
+        default_timezone=timezone.utc,
+        required=False,
+    )
+    end = serializers.DateTimeField(
+        format=date_format, default=None, default_timezone=timezone.utc, required=False
+    )
+
+    def validate(self, data):
+        if "start" in data and "end" in data and data["start"] > data["end"]:
+            raise serializers.ValidationError("Invalid timestamp (start must be before end).")
+        return data
+
+    def validate_end(self, end):
+        if end is None:
+            end = datetime.now(tz=timezone.utc)
+        return end
+
+
+def filter_by_date(request: RpcSentryAppRequest, start: datetime, end: datetime) -> bool:
+    date_str = request.date
+    if not date_str:
+        return False
+    timestamp = datetime.strptime(date_str, "%Y-%m-%d %H:%M:%S.%f+00:00").replace(
+        microsecond=0, tzinfo=timezone.utc
+    )
+    return start <= timestamp <= end
+
+
+def filter_by_organization(
+    request: RpcSentryAppRequest, organization: OrganizationMapping | None
+) -> bool:
+    if not organization:
+        return True
+    return request.organization_id == organization.organization_id
```

Comment:
(nit) these can be private methods on the endpoint class if they're only used there.